pulser-core/pulser/register/register.py

```diff
@@ -448,25 +452,56 @@ def draw(
             draw_half_radius=draw_half_radius,
         )
 
+        if draw_empty_sites:
+            if self.layout is None:
+                raise ValueError(
+                    "The register must have an associated RegisterLayout "
+                    "to draw the empty sites."
+                )
+            layout_ids = list(self.layout.traps_dict.keys())
+            empty_layout = self.layout.define_register(
+                *layout_ids, qubit_ids=layout_ids
+            )
+            breakpoint()
+            empty_qubit_colors : Mapping[Union[int, str], str] = {
+                trap: empty_color or "r" for trap in layout_ids
+            }
+            empty_pos = empty_layout._coords_arr.as_array(detach=True)
+
         pos = self._coords_arr.as_array(detach=True)
         if custom_ax is None:
             custom_ax = cast(
                 plt.Axes,
                 self._initialize_fig_axes(
-                    pos,
+                    empty_pos if draw_empty_sites else pos,
                     blockade_radius=blockade_radius,
-                    draw_half_radius=draw_half_radius,
-                )[1],
-            )
-        super()._draw_2D(
-            custom_ax,
-            pos,
-            self._ids,
-            with_labels=with_labels,
+                    draw_half_radius=draw_half_radius
+                    )[1],
+                )
+
+        draw_kwargs = dict(
+            ax=custom_ax,
             blockade_radius=blockade_radius,
             draw_graph=draw_graph,
             draw_half_radius=draw_half_radius,
+        )
+
+        if draw_empty_sites:
+            RegDrawer._draw_2D(empty_layout,
+                ids=empty_layout._ids,
+                pos=empty_pos,
```

Comment:
Your implementation looks good. It enables you to implement the alternative I have presented above, where the empty spots are circles filled with white. To implement it fully, you might want to take out the positions of the atoms in the Register in empty_pos, because at the moment you are super-imposing two symbols for these filled positions and the symbol of the circle with white inside is larger than the symbol for the atoms (a disk of green by default, or a color defined in qubit_colors).